### ISSUE TITLE
Disabled button state doesn't use native semantics

### ISSUE THREAD

**Issue by anselmbradford** [cfpb/capital-framework#546]
We have `a-btn__disabled` class, which in the docs is applied to a button that doesn't have the `disabled` attribute set. Shouldn't we use the native semantics and set the `disabled` attribute on the button, ditch the custom class name, and style it accordingly? 

## Current behavior

- `a-btn__disabled` class is used in place of `disabled` attribute.

## Expected behavior

- Native `disabled` attribute could be applied and style the button according to the `a-btn__disabled` style.

**Comment by Scotchester** [comment]
This would work for `input` and `button` elements, but not for button-styled `a` elements. Perhaps we should support both.

**Comment by anselmbradford** [comment]
Ah right, yeah we should probably support both.

**Comment by jimmynotjim** [comment]
Like making `a-text-input` I believe we used a class specifically to avoid having to figure out all the attribute combinations when overriding styles. The example should be using the `disabled` attribute in the markup, but only to allow the browser to handle the interaction, not as a styling hook.

**Comment by anselmbradford** [comment]
@jimmynotjim Sorry, what did you mean with this? Is `a-text-input__disabled` needed? Or is that it was added to clarify things, but wouldn't be needed?

**Comment by jimmynotjim** [comment]
Sorry, I completely botched that comment. 

I was trying to explain why we used `a-btn` to capture all element types instead of something like `a.btn, button.btn, input.btn` similar to why we use `a-text-input`. I didn't mean it to sound like `a-text-input__disabled` and `a-btn__disabled` had any relationship.

The other half of my comment I was trying to say that I believe we should continue using `a-btn__disabled ` as the single styling hook, for the reasons that @Scotchester pointed out about links, but that the markup should still get the attribute for whatever native browser interactions exist outside styles. 

TL;DR I don't think we should do `a-btn[disabled]` because it won't work for links and having two patterns is just one more thing to remember.

**Comment by jimmynotjim** [comment]
I believe the `disabled` attributes were all added to the docs some time ago. I think we can close this.
